plume/src/main/kotlin/io/github/plume/oss/passes/type/GlobalTypePass.kt

```diff
@@ -55,7 +66,15 @@ class GlobalTypePass(private val driver: IDriver) : ITypePass {
                 .order(-1)
                 .filename("<unknown>")
                 .astParentType(NAMESPACE_BLOCK)
-                .astParentFullName("<global>").apply { nodeCache.add(this) }
+                .astParentFullName("<global>").apply {
+                    val type = NewTypeBuilder().name(shortName)
```

Comment:
Mind moving this into a method similar to `getGlobalTypeDecl` and call it `getGlobalType` which can then look something like
```kotlin
private fun getGlobalType(tdFullName: String): NewNodeBuilder {
       val shortName = if (tdFullName.contains('.')) tdFullName.substringAfterLast('.')
       else tdFullName
       return nodeCache.filterIsInstance<NewTypeBuilder>()
            .find { it.build().properties().get(FULL_NAME).get() == tdFullName }
            ?: NewTypeBuilder()
                            .name(shortName)
                            .fullName(tdFullName)
                            .typeDeclFullName(tdFullName)
                            .apply { nodeCache.add(this) }
}
```
Then under `runPass` you can change
```kotlin
    .forEach { t ->
                logger.debug("Upserting and linking for global type ${t.build().properties()[FULL_NAME]}")
                driver.addEdge(n, t, AST)
                driver.addEdge(t, f, SOURCE_FILE)
                driver.addEdge(f, t, CONTAINS)
    }
```
to
```kotlin
    .forEach { td ->
                val tdFullName = t.build().properties()[FULL_NAME]
                logger.debug("Upserting and linking for global type ${tdFullName}")
                val t = getGlobalType(tdFullName)
                driver.addEdge(n, td, AST)
                driver.addEdge(td, f, SOURCE_FILE)
                driver.addEdge(fd, t, CONTAINS)
                driver.addEdge(t, td, REF)
    }
```